### PR TITLE
Catch all port configurations 

### PR DESCRIPTION
**Pull request by Fenn-CS**
Added other port config formats for port 0

---

Ticket [trac]: [_tub_portlocation tries to detect port 0 configuration but does so incompletely](https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3563 )
Also closes [trac]: [Intermittent test failure: allmydata.test.test_node.TestMissingPorts.test_listen_on_zero_with_host](https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3776)

### CODE REVIEW

File: newsfragments/3563.minor [hunk 1]
```diff
@@ -0,0 +1 @@
+detect all port 0 configs
```
Comment (itamarst) [inline newsfragments/3563.minor:1]: My understanding of the convention is that if it's minor enough to merit a `.minor`, you can just leave the file empty. And it's meaningful enough to have a description, should maybe be `.bugfix` and be a more fleshed out English sentence that's meaningful to users. This seems pretty minor, so I'd do the former.

**Review by itamarst — APPROVED** [review]
Thank you!

File: newsfragments/3563.minor (hunk 1 above)
Comment (Fenn-CS) [reply newsfragments/3563.minor:1]: Thanks noted. Do I have to undo this in this current PR?

Comment (itamarst) [reply newsfragments/3563.minor:1]: Probably fine?

Comment (Fenn-CS) [reply newsfragments/3563.minor:1]: Great removing that

**Comment by Fenn-CS** [comment]
@exarkun @May-Lee @meejah We have to merge this for other PRs to pass.

**Pull request merged**